### PR TITLE
651 - CLI - deployment and installation script #651

### LINKED ISSUE #651: CLI - deployment and installation script
### Suggestion

pipeline to release and deploy the CLI executable for each OS.
also, installation script to install confifu CLI install Linux / mac / docker env.

### Motivation

_No response_

### Context

_No response_

### PR DESCRIPTION
**Pull request by davidmeirlevy**
Closes #651

### CODE REVIEW

File: .github/workflows/cd-cli.yml [hunk 1]
```diff
@@ -36,33 +43,55 @@ jobs:
...
-          tag_name: cli-${{ steps.deployment.outputs.release-version }}
+          make_latest: ${{  github.branch == 'main' }}
+          prerelease: ${{  github.branch != 'main' }}
+          name: ${{  github.ref_name }}
+          tag_name: ${{  github.ref_name }}
           files: |
             ${{ env.ARTIFACT_NAME }}/dist/*
+
+  docker:
+    runs-on: ubuntu-latest
```
Comment (pelegpor) [inline .github/workflows/cd-cli.yml:82]: It should be `ubuntu:latest`, no? @davidmeirlevy @rannn505

Comment (pelegpor) [reply .github/workflows/cd-cli.yml:82]: https://github.com/orgs/community/discussions/31587

File: packages/cli/install.sh [hunk 2]
```diff
@@ -0,0 +1,54 @@
+#!/bin/sh
+
+set -e
+
+if [ "$OS" = "Windows_NT" ]; then
+	target="win32-x64"
+	ext="exe"
```
Comment (rannn505) [inline packages/cli/install.sh:7]: why are you using tabs in this script?
why the windows assertion is needed here? isnt it for bash envs?

Comment (davidmeirlevy) [reply packages/cli/install.sh:7]: there's bash for windows, and there are multiple ways to work with bash on windows:
- when installing git on windows -> bash is installed by default.
- "windows terminal"  (an official Microsoft terminal - https://apps.microsoft.com/detail/9n0dx20hk701?hl=en-US&gl=US) supports working with bash.
- WSL

Comment (davidmeirlevy) [reply packages/cli/install.sh:7]: actually, from a WSL terminal - PowerShell will not work.

File: packages/cli/compiler/darwin.ts [hunk 3]
```diff
@@ -11,4 +14,6 @@ export function run(filename = 'configu') {
   );
   execSync('codesign --sign - configu');
```
Comment (rannn505) [inline packages/cli/compiler/darwin.ts:15]: image

 I think this is not working

File: packages/cli/compiler/gz.ts [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import { createGzip } from 'node:zlib';
+import { createReadStream, createWriteStream } from 'node:fs';
+import { pipeline } from 'node:stream/promises';
+
+export async function compress(input) {
```
Comment (rannn505) [inline packages/cli/compiler/gz.ts:5]: im not a fan of utilising the gzipped executables as it adds another fail point to the install script

File: packages/cli/compiler/win32.ts [hunk 5]
```diff
@@ -1,14 +1,60 @@
 import { execSync } from 'node:child_process';
```
Comment (rannn505) [inline packages/cli/compiler/win32.ts:1]: why didnt you use $ from zx?
import { $ } from 'zx';

Comment (davidmeirlevy) [reply packages/cli/compiler/win32.ts:1]: i didn't want to add more dependencies.
used only native operations for now.
we can improve it later.

**Review by rannn505 — APPROVED** [review]

File: packages/cli/compiler/win32.ts (hunk 5 above)
Comment (rannn505) [reply packages/cli/compiler/win32.ts:1]: it is already installed on the global monorepo as dev dep

**Pull request merged**